caikit_nlp/modules/text_generation/peft_tgis_remote.py

```diff
@@ -221,6 +233,9 @@ def run(
             self.enable_backend,
             "Backend must be configured and loaded with this module before executing `run` call.",
         )
+        if self._tgis_backend:
```

Comment:
Total NIT, but since `_register_model_connection_with_context` is a member function, you could move the `if self._tgis_backend` line into the function and save a few duplicate lines in each of the inference methods.